src/Notepads/Core/NotepadsCore.cs

```diff
@@ -798,15 +799,46 @@ private async void Sets_SetDraggedOutside(object sender, SetDraggedOutsideEventA
         {
             if (Sets.Items?.Count > 1 && e.Set.Content is ITextEditor textEditor)
             {
-                // Only allow untitled empty document to be dragged outside for now
-                if (!textEditor.IsModified && textEditor.EditingFile == null)
+                if (textEditor.FileModificationState != FileModificationState.RenamedMovedOrDeleted)
                 {
                     DeleteTextEditor(textEditor);
```

Comment:
Don't delete right away, the reason I put delete first was because the editor must be empty. Now we have value, so make we delete it after LaunchProtocolAsync returns true. You can also first make the eidtor.IsEnabled = false, and LaunchProtocolAsync, set it to true or deleted based on result.